redbeams/src/main/java/com/sequenceiq/redbeams/service/UserGeneratorService.java

```diff
@@ -13,14 +17,34 @@
 
     private static final int USER_NAME_LENGTH = 10;
 
+    private static final String UUID_EXTRACTOR_REGEXP = "(\\w{8})-(\\w{4})-\\w(\\w{3})-\\w(\\w{3})-(\\w{12})";
+
+    private static final int MAX_LENGTH_PASSWORD = 30;
+
+    @Inject
+    private UUIDGeneratorService uuidGeneratorService;
+
     public String generateUserName() {
         return ThreadLocalRandom.current().ints(0, MAX_RANDOM_INT_FOR_CHARACTER)
                 .limit(USER_NAME_LENGTH).boxed()
                 .map(i -> Character.toString((char) ('a' + i)))
                 .collect(Collectors.joining());
     }
 
+    /*
+    Generates a password using a UUID.
+    Provider specific limitations:
+    - AWS: password length in cf template can be 30 characters long at most
+     */
     public String generatePassword() {
-        return UUID.randomUUID().toString();
+        Pattern pattern = Pattern.compile(UUID_EXTRACTOR_REGEXP);
```

Comment:
Instead of compiling the pattern each time this method is called, just compile it once and make it a static final. Pattern objects are safe to share across multiple threads.